models/Asset/Listing/Dao.php

```diff
@@ -92,11 +92,15 @@ public function getCount(): int
 
     public function getTotalCount(): int
     {
+        $identifierColumn = 'assets.id';
         $queryBuilder = $this->getQueryBuilder();
-        $this->prepareQueryBuilderForTotalCount($queryBuilder, 'assets.id');
+        $this->prepareQueryBuilderForTotalCount($queryBuilder, $identifierColumn);
 
-        $amount = (int) $this->db->fetchOne($queryBuilder->getSql(), $queryBuilder->getParameters(), $queryBuilder->getParameterTypes());
-
-        return $amount;
+        if ($this->isQueryBuilderPartInUse($queryBuilder, 'groupBy') || $this->isQueryBuilderPartInUse($queryBuilder, 'having')) {
+            $queryBuilder->select($identifierColumn);
```

Comment:
```suggestion
            if (!$this->isQueryBuilderPartInUse($queryBuilder, 'select')){
               $queryBuilder->select($identifierColumn);
            }
```
for some reason i don't know why, this wouldn't pass codeception tests, but locally seems working fine... 
the idea is to replicate the former 
https://github.com/pimcore/pimcore/pull/18150/files#diff-09de6410fb1e06d3d60e14877c348e74e247f47b13a4f4730bf76b9ff19bf744L143
leave unaltered when is already set, if not set then use identifier

Reply:
tests are green (beside sonarcloud) on https://github.com/pimcore/pimcore/pull/18150/commits/fd2038403ef0c92bd8e76d4c2d3e3becd1dea4b2 , let's see https://github.com/pimcore/pimcore/pull/18150/commits/25413544a7a882443e14b487f80cae3cbd1c7325 after applying this change